### ISSUE TITLE
panda-test fails; Currently no JVM Proc::Async

### ISSUE THREAD

**Issue by ugexe** [tadzik/panda#181]
perl6-j -e 'my $proc = Proc::Async;'
    Could not find symbol '&Async'
      in block <unit> at -e:1

**Comment by FROGGS** [comment]
This came in because of #179...

@hoelzro How much work would be needed to switch to `shell(:out, :err)` ?
Or is there another option?

**Comment by hoelzro** [comment]
The reason I diverged from using `shell` and company is because it seemed difficult to read from standard output and standard error at the same time.  Is there a way one can do this with `shell`?

**Comment by ugexe** [comment]
perl6 -e 'my $proc = run("$*EXECUTABLE", "-e", "say 1; die", :out, :err); say "out: {$proc.out.lines}"; say "err: {$proc.err.lines}"'
    out: 1
    err: Died   in block <unit> at -e:1

**Comment by hoelzro** [comment]
@ugexe Can we get the interleaved lines, though? Ex. if I do this:

```perl6
$*OUT.say: 1;
$*ERR.say: 2;
$*OUT.say: 3;
```

**Comment by ugexe** [comment]
perl6 -e 'my $proc = run("$*EXECUTABLE", "-e", "say 1; warn 2; say 3;", :out, :err); say "out: {$proc.out.lines.perl}"; say "err: {$proc.err.lines.perl}"'
    out: ("1", "3", "")
    err: ("2  in block <unit> at -e:1", "")

**Comment by FROGGS** [comment]
you read it out of order then...

**Comment by FROGGS** [comment]
Does that mean that we want to port Proc::Async to JVM real quick?

**Comment by hoelzro** [comment]
That, or have Panda fall back to its old behavior when `Proc::Async` is not available.

**Comment by Leont** [comment]
> @ugexe Can we get the interleaved lines, though? Ex. if I do this:

That's not how pipes work. Different pipe have no inherent order between each other, only inside the stream. The only way to synchronize it is to merge them (I'm not sure we have a syntax for that), but that is not necessarily the desired behavior.

**Comment by hoelzro** [comment]
@Leont I think using something like `select` would be sufficient for our needs.

**Comment by Leont** [comment]
> I think using something like select would be sufficient for our needs.

Mostly, yes. The point I'm trying to make is that merging them later on gives no guarantees about ordering, it may even mix lines from different streams. The concept seems misguided to me.

Having them separate and keeping them separate is perfectly sane OTOH. Until the pipe buffer is filled up you could do even without a `select`, but you can be sure someone someday will sent large amounts of data to stderr and you don't want to be screwed by that.

**Comment by hoelzro** [comment]
I don't think it makes sense to try and merge them either; I'm just trying to preserve existing behavior until tools that rely on the old behavior have had a chance to migrate.
